Added ability to do segmented peak detection on polar transform of DP

NJMarchese opened the pull request:
- Added new function to allow for different peak detection parameters for different q-spacings on the diffraction pattern. 
- Also enabled the ability to pass a realspace mask onto find_peaks. 
- Added an error catch onto the curve_fit of flowlines to give a more informative error message.
- Added kwargs.get method to find_peaks_single_pattern so that the figure and axes are passed correctly. Before wouldn't handle correctly, and if figax was passed it would give an error saying there were multiple values for figax (the default in the code and the one passed in kwargs)
- Added vmin and vmax kwargs pass to show_amorphous_ring_fit
- Added ability to return data for plot_radial_peaks

NJMarchese committed:
Added ability to do segmented peak detection on polar transform of DP

NJMarchese committed:
Re-formatted previous commit's files with black linter

smribet commented on py4DSTEM/visualize/vis_special.py:
```diff
@@ -159,6 +159,10 @@ def show_amorphous_ring_fit(
         **kwargs,
     )
+    if kwargs.get("vmax") is not None:
+        vmax = kwargs.pop("vmax")
```
I think if you want to do this you need add vmin and max as flags to the function

NJMarchese replied:
Changed to add vmin and vmax as flags for the function input, but also had to change other lines of code since variable with names vmin and vmax assigned in line 154.

smribet commented on py4DSTEM/process/polar/polar_datacube.py:
```diff
@@ -115,6 +115,9 @@ def __init__(
         plot_radial_background,
         model_radial_background,
         make_orientation_histogram,
+        # Added
```
Please remove any commented out code that's specific to your PR

smribet commented on py4DSTEM/process/polar/polar_peaks.py:
```diff
@@ -321,7 +325,8 @@ def find_peaks_single_pattern(
-            s=peaks_polar["intensity"] * plot_scale_size,
+            # s=peaks_polar["intensity"] * plot_scale_size,
+            s=10,
             marker="o",
```
In general, I'd suggest leaving this more flexible in case someone has different plot sizes than you

NJMarchese replied:
Returned to the original line of code for marker size

smribet commented on py4DSTEM/process/polar/polar_peaks.py:
```diff
@@ -472,6 +760,153 @@ def find_peaks(
+    kwargs_pass=None,
+):
+    """
+    Peak detection function for polar transformations. Loop through all probe positions,
```
missing doc strings...especially for your new contributions with kwargs_pass, it'd be helpful to add detailed docs here!

NJMarchese replied:
I put in some doc strings, but as this is built off the original find_peaks function and the docstrings for that function is what is in find_peaks_segmented, I'm not sure if it's best for me to fill in those belonging to the original function. I added docstrings for the parameters I added for find_peaks_segmented.

smribet commented on py4DSTEM/process/polar/polar_peaks.py:
```diff
@@ -472,6 +760,153 @@ def find_peaks(
+def find_peaks_segmented(
+    self,
+    mask=None,
+    mask_real=None,
```
Can this be added to the other find_peaks function too?

NJMarchese replied:
Maybe. When I looked at it the first time it seemed like it would break a bunch of things, so made it separate initially to keep everything in tact. I can take another look.

smribet commented:
Hi @NJMarchese, thanks for contributing these features! I have a few small suggestions to your code.

NJMarchese committed:
Refactor peak plotting and add mask parameters for peak detection

- Updated `find_peaks_single_pattern` to restore intensity scaling for scatter plot.
- Enhanced `find_peaks_segmented` with new mask parameters for improved peak detection.
- Modified `plot_radial_peaks` to allow customizable vertical lines.
- Adjusted orientation histogram calculation to ensure angle conversion is in radians.
- Added vmin and vmax parameters to `show_amorphous_ring_fit` for better visualization control.
